File: backend/entitlements.py
import os
"""Central feature-entitlement configuration. Single source of truth for plan gating."""

PLAN_ORDER = ["free", "start", "grow", "pro"]

PLANS = {
    "free": {
        "id": "free",
        "name": "FREE",
        "price_bdt": 0,
        "recommended": False,
        "tagline": "Try Nexora and make your first sales",
        "commission_percent": 10,
        "entitlements": {
            "max_products": 10,
            "theme_switching": False,
            "theme_access": "essential",
            "custom_accent_color": False,
            "custom_css": False,
            "collections_limit": 1,
            "sections_limit": 4,
            "advanced_analytics": False,
            "bulk_management": False,
            "marketing_tools": False,
            "customer_insights": False,
            "abandoned_cart": False,
            "staff_accounts": False,
            "ai_features": False,
            "priority_support": False,
            "marketplace_intelligence": False,
            "demand_analysis": False,
            "competitor_benchmark": False,
            "ai_product_suggestions": False,
            "ai_pricing": False,
            "store_import": False,
            "auto_store_sync": False,
            "transaction_fee": 10,
        },
    },
    "start": {
        "id": "start",
        "name": "START",
        "price_bdt": 500,
        "recommended": False,
        "tagline": "Start selling",
        "commission_percent": 7,
        "entitlements": {
            "max_products": 50,
            "theme_switching": False,
            "theme_access": "essential",
            "custom_accent_color": False,
            "custom_css": False,
            "collections_limit": 2,
            "sections_limit": 4,
            "advanced_analytics": False,
            "bulk_management": False,
            "marketing_tools": False,
            "customer_insights": False,
            "abandoned_cart": False,
            "staff_accounts": False,
            "ai_features": False,
            "priority_support": False,
            "marketplace_intelligence": False,
            "demand_analysis": False,
            "competitor_benchmark": False,
            "ai_product_suggestions": False,
            "ai_pricing": False,
            "store_import": False,
            "auto_store_sync": False,
            "transaction_fee": 7,
        },
    },
    "grow": {
        "id": "grow",
        "name": "GROW",
        "price_bdt": 1500,
        "recommended": True,
        "tagline": "Grow your business",
        "commission_percent": 5,
        "entitlements": {
            "max_products": 500,
            "theme_switching": True,
            "theme_access": "all",
            "custom_accent_color": True,
            "custom_css": False,
            "collections_limit": 10,
            "sections_limit": 10,
            "advanced_analytics": True,
            "bulk_management": True,
            "marketing_tools": True,
            "customer_insights": True,
            "abandoned_cart": True,
            "staff_accounts": True,
            "ai_features": False,
            "priority_support": False,
            "marketplace_intelligence": False,
            "demand_analysis": False,
            "competitor_benchmark": False,
            "ai_product_suggestions": False,
            "ai_pricing": False,
            "store_import": False,
            "auto_store_sync": False,
            "transaction_fee": 5,
        },
    },
    "pro": {
        "id": "pro",
        "name": "PRO",
        "price_bdt": 3000,
        "recommended": False,
        "tagline": "Nexora Intelligence + automation",
        "commission_percent": 3,
        "entitlements": {
            "max_products": -1,
            "theme_switching": True,
            "theme_access": "premium",
            "custom_accent_color": True,
            "custom_css": False,
            "collections_limit": -1,
            "sections_limit": -1,
            "advanced_analytics": True,
            "bulk_management": True,
            "marketing_tools": True,
            "customer_insights": True,
            "abandoned_cart": True,
            "staff_accounts": True,
            "ai_features": True,
            "priority_support": True,
            "marketplace_intelligence": True,
            "demand_analysis": True,
            "competitor_benchmark": True,
            "ai_product_suggestions": True,
            "ai_pricing": True,
            "store_import": True,
            "auto_store_sync": True,
            "transaction_fee": 3,
        },
    },
}

# Keep the current project's existing behaviour.
PLANS["grow"]["entitlements"]["theme_switching"] = False

for plan_id, plan in PLANS.items():
    plan["price_bdt"] = int(os.getenv(f"PLAN_{plan_id.upper()}_BDT", str(plan["price_bdt"])))
    plan["billing_period"] = os.getenv("BILLING_PERIOD") or None
    plan["billing_enabled"] = False
    plan["entitlements"]["logo_upload"] = True
    plan["entitlements"]["cover_upload"] = True
    plan["entitlements"]["layout_customization"] = plan_id == "pro"

PLAN_FEATURES = {
    "free": [
        "Category-specific starter storefront",
        "Up to 10 products",
        "Inventory and order management",
        "Courier-ready seller setup",
        "10% marketplace commission",
    ],
    "start": [
        "Category-specific preset",
        "Logo and cover uploads",
        "Up to 50 products",
        "Inventory and orders",
    ],
    "grow": [
        "Logo and cover uploads",
        "Up to 500 products",
        "Limited brand color customization",
        "Bulk catalogue management",
        "Advanced analytics",
        "Customer insights",
        "Marketing tools",
    ],
    "pro": [
        "Logo and cover uploads",
        "Unlimited products",
        "Theme selection",
        "Full layout, font and color controls",
        "Advanced analytics",
        "Nexora Marketplace Intelligence",
        "Demand vs supply opportunity analysis",
        "Anonymous marketplace pricing benchmarks",
        "AI-assisted recommendations",
        "Import an existing website",
        "Automatic product sync",
        "Priority support",
    ],
}


def get_plan(plan_id: str) -> dict:
    return PLANS.get(plan_id, PLANS["start"])


def get_entitlements(plan_id: str) -> dict:
    return get_plan(plan_id)["entitlements"]

File: frontend/src/lib/entitlements.js
// Frontend mirror of backend entitlements for instant UI gating.
export const PLAN_META = {
  free: { id: "free", name: "FREE", price: 0, color: "#0F8A68", tagline: "Try Nexora" },
  start: { id: "start", name: "START", price: 500, color: "#66736B", tagline: "Start selling" },
  grow: { id: "grow", name: "GROW", price: 1500, color: "#10B981", tagline: "Grow your business" },
  pro: { id: "pro", name: "PRO", price: 3000, color: "#F59E0B", tagline: "Nexora Intelligence" },
};

export const PLAN_ORDER = ["free", "start", "grow", "pro"];

export function planRank(plan) {
  return PLAN_ORDER.indexOf(plan);
}

export function can(entitlements, feature) {
  if (!entitlements) return false;
  return !!entitlements[feature];
}

export const FEATURE_LABELS = {
  theme_switching: "Switch storefront themes",
  custom_accent_color: "Custom brand colors",
  custom_css: "Custom CSS styling",
  advanced_analytics: "Advanced analytics",
  bulk_management: "Bulk catalog management",
  marketing_tools: "Marketing tools",
  customer_insights: "Customer insights",
  abandoned_cart: "Abandoned cart recovery",
  staff_accounts: "Staff accounts",
  ai_features: "AI feature suite",
  priority_support: "Priority support",
  marketplace_intelligence: "Nexora Marketplace Intelligence",
  demand_analysis: "Marketplace demand analysis",
  competitor_benchmark: "Anonymous marketplace benchmarking",
  ai_product_suggestions: "AI product opportunity suggestions",
  ai_pricing: "AI pricing intelligence",
  store_import: "Import an existing website",
  auto_store_sync: "Automatic store sync",
};

export function requiredPlanFor(feature) {
  const grow = [
    "theme_switching",
    "custom_accent_color",
    "advanced_analytics",
    "bulk_management",
    "marketing_tools",
    "customer_insights",
    "abandoned_cart",
    "staff_accounts",
  ];
  const pro = [
    "custom_css",
    "ai_features",
    "priority_support",
    "marketplace_intelligence",
    "demand_analysis",
    "competitor_benchmark",
    "ai_product_suggestions",
    "ai_pricing",
    "store_import",
    "auto_store_sync",
  ];
  if (pro.includes(feature)) return "pro";
  if (grow.includes(feature)) return "grow";
  return "free";
}
